### PR TITLE
feat: write span in batch to redis

### PR DESCRIPTION
**Pull request by VinozzZ**
## Which problem is this PR solving?

To increase throughput for writing spans into redis, we should write spans in batches instead of one by one.

## Short description of the changes

- Change `WriteSpan` to `WriteSpans`
- use transaction for independent redis calls for `addStatus`
- Defined a constant batch size of 10 as a starting point for WriteSpans

### CODE REVIEW

File: centralstore/local_store.go [hunk 1]
```diff
@@ -123,69 +123,70 @@ func (lrs *LocalStore) changeTraceState(traceID string, fromState, toState Centr
...
 	lrs.mutex.Lock()
 	defer lrs.mutex.Unlock()
 
-	// first let's check if we've already processed and dropped this trace; if so, we're done and
-	// can just ignore the span.
-	if lrs.DecisionCache.Dropped(span.TraceID) {
-		otelutil.AddSpanField(spanWrite, "dropped", true)
-		return nil
-	}
+	for _, span := range spans {
```
Comment (VinozzZ) [inline centralstore/local_store.go:133]: I just moved all the logic into a loop here

**Comment by VinozzZ** [comment]
It might be good to check for performance before we merge this.

File: centralstore/smartwrapper.go [hunk 2]
```diff
@@ -113,8 +115,24 @@ func (w *SmartWrapper) WriteSpan(ctx context.Context, span *CentralSpan) error {
...
-	for span := range w.spanChan {
-		err := w.BasicStore.WriteSpan(ctx, span)
+	for first := range w.spanChan {
+		var spans []*CentralSpan
+		spans = append(spans, first)
+
+	Remaining:
+		for len(spans) < WriteSpanBatchSize { // For control maximum size of batch
+			select {
+			case span, ok := <-w.spanChan:
```
Comment (TylerHelmuth) [inline centralstore/smartwrapper.go:125]: Will this have an opposite reaction where refinery will pause on processing spans when the rate of incoming spans is slow and it takes a while to reach `WriteSpanBatchSize`?

**Comment by kentquirk** [comment]
I added my `exists` tweak to this PR.

**Review by kentquirk — APPROVED** [review]
I think this is ready to go now.

**Pull request merged**